testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/MultinodeKafka.java

```diff
@@ -20,27 +20,28 @@
 import io.trino.tests.product.launcher.env.Environment;
 import io.trino.tests.product.launcher.env.EnvironmentProvider;
 import io.trino.tests.product.launcher.env.common.Kafka;
-import io.trino.tests.product.launcher.env.common.Standard;
 import io.trino.tests.product.launcher.env.common.TestsEnvironment;
+import io.trino.tests.product.launcher.env.common.TrinoMultinode;
 
 import javax.inject.Inject;
 
 import static io.trino.tests.product.launcher.env.EnvironmentContainers.COORDINATOR;
 import static io.trino.tests.product.launcher.env.EnvironmentContainers.TESTS;
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.WORKER;
 import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_PRESTO_ETC;
 import static java.util.Objects.requireNonNull;
 import static org.testcontainers.utility.MountableFile.forHostPath;
 
 @TestsEnvironment
-public final class SinglenodeKafka
+public final class MultinodeKafka
```

Comment:
@Praveen2112 @kokosing please help me understand. Why did the problem not manifiset with single node test env?

Reply:
When `SchemaRegistryClient` tries to invoke `UriBuilder.fromPath` - which in turn create a new Instance of `UriBuilder` for the first time. In case of coordinator this happens inside `KafkaMetadata` which is wrapped with its classloader based implementation - so single node test env doesn't throw this error 

This issue is seen in Worker where `UriBuilder.fromPath` is invoked for the first time in `KafkaRecordCursor#nextRow` which is under lib's classloader and it throws Class cast exception.